synapse/http/server.py

```diff
@@ -353,17 +353,23 @@ def render(self, request):
         """
         Render the request, using an asynchronous render handler if it exists.
         """
-        render_callback_name = "_async_render_" + request.method.decode("ascii")
+        render_callback_name = "render_" + request.method.decode("ascii")
+        async_render_callback_name = "_async_" + render_callback_name
 
-        if hasattr(self, render_callback_name):
+        if hasattr(self, async_render_callback_name):
```

Comment:
non-blocking thought: given that `hasattr` is implemented via `getattr` (https://docs.python.org/3/library/functions.html#hasattr), it might be nice to reorder this to save the double-lookup.

Reply:
yeah cleaned that up, fixed